JPetTaskInterface/JPetTaskInterface.h

```diff
@@ -33,8 +35,8 @@ class JPetTaskInterface
   virtual bool init(const JPetParamsInterface& inOptions) = 0;
   virtual bool run(const JPetDataInterface& inData) = 0;
   virtual bool terminate(JPetParamsInterface& outOptions) = 0;
-  virtual void setSubTask(std::unique_ptr<JPetTaskInterface> subTask) = 0;
-  virtual JPetTaskInterface* getSubTask() const = 0;
-  virtual std::string getName() const =0;
+  virtual void addSubTask(std::shared_ptr<JPetTaskInterface> subTask) = 0;
+  virtual std::vector<std::shared_ptr<JPetTaskInterface>> getSubTasks() const = 0;
```

Comment:
Here * was used to express the fact that the caller does not take the ownership of the task.